pandas/tools/pivot.py

```diff
@@ -159,6 +159,19 @@ def _add_margins(table, data, values, rows, cols, aggfunc):
 
     grand_margin = _compute_grand_margin(data, values, aggfunc)
 
+    # categorical index or columns will fail below when 'All' is added
+    # here we'll convert all categorical indices to object
+    def convert_categorical(ind):
+        _convert = lambda ind: (ind.astype('object')
+                                if ind.dtype.name == 'category' else ind)
```

Comment:
The test case is in the linked issue. Or would you like me to add a unit test to the package as part of this PR?